lib/cell.rb

```diff
@@ -1,12 +1,51 @@
 class Cell
 
-  attr_reader :coordinates
+  attr_reader :coordinates,
+              :ship,
+              :fired_upon
 
   def initialize(coordinates)
     @coordinates = coordinates
+    @ship = nil
+    @fired_upon = false
+
+  end
+
+  def empty?
+    if @ship.class == Ship
+      false
+    else
+      true
+    end
+  end
+
+  def place_ship(ship)
+    @ship = ship
+  end
+
+  def fired_upon?
+    @fired_upon
+  end
+
+  def fire_upon
+    if @ship.class == Ship
+      @fired_upon = true
+      @ship.hit
+    else
+    end
   end
 
-  def render
 
+  def render(is_there_a_ship = false)
+    if @fired_upon = false
+      "."
+    #elsif @fired_upon = true || @ship.hit = false
```

Comment:
@amymoller you have the right idea!!! 
Couple things - in conditional statements you need to use `==` instead of `=` because you're _comparing_ the things on either side of the `==` and _not_ assigning a new variable (`=`)

The other thing is that I see why you want to use `@ship.hit` in there, but if we look at what the #hit method does inside the `Ship` class, it's not a boolean method, aka, it will never be `true` or `false`, so we want to find a different way to check that the ship was hit. _(Not sure what to use yet, gotta think on it)_

Reply:
Oh, and yes to your comment on slack about using `&&` instead of `||`

Reply:
note from Tim: make the "." default render as the last 'else' piece in the conditional.